test/formatters/ecs_test.exs

```diff
@@ -0,0 +1,381 @@
+        Logger.debug("oops!")
+      end)
+      |> decode_or_print_error()
+
+    assert %{
+             "message" => "oops!",
+             "error.message" => "exit: :sad_failure",
+             "error.stack_trace" => "** (exit) :sad_failure",
+             "error.type" => "exit",
+             "log.logger" => "Elixir.LoggerJSON.Formatters.ECSTest",
+             "log.origin" => %{
+               "file.line" => _,
+               "file.name" => _,
+               "function" => _
+             }
+           } = log_entry
+  end
+
+  test "logs process exits" do
+    Logger.metadata(crash_reason: {{:EXIT, self()}, :sad_failure})
+
+    log_entry =
+      capture_log(fn ->
+        Logger.debug("oops!")
+      end)
+      |> decode_or_print_error()
+
+    assert %{
+             "message" => "oops!",
+             "error.message" => error_message,
+             "error.stack_trace" => stacktrace,
+             "error.type" => "EXIT",
+             "log.logger" => "Elixir.LoggerJSON.Formatters.ECSTest",
+             "log.origin" => %{
+               "file.line" => _,
+               "file.name" => _,
+               "function" => _
+             }
+           } = log_entry
+
+    assert stacktrace =~ ~r/\*\* \(EXIT from #PID<\d+\.\d+\.\d+>\) :sad_failure/
+    assert error_message =~ ~r/process #PID<\d+\.\d+\.\d+> exit: :sad_failure/
+  end
+
+  test "logs reasons in tuple" do
+    Logger.metadata(crash_reason: {:socket_closed_unexpectedly, []})
+
+    log_entry =
+      capture_log(fn ->
+        Logger.debug("oops!")
+      end)
+      |> decode_or_print_error()
+
+    assert %{
+             "message" => "oops!",
+             "error.message" => "socket_closed_unexpectedly: []",
+             "error.stack_trace" => "** (socket_closed_unexpectedly) []",
+             "error.type" => "socket_closed_unexpectedly",
+             "log.logger" => "Elixir.LoggerJSON.Formatters.ECSTest",
+             "log.origin" => %{
+               "file.line" => _,
+               "file.name" => _,
+               "function" => _
+             }
+           } = log_entry
+  end
+
+  test "logs http context" do
+    conn =
+      Plug.Test.conn("GET", "/", "")
+      |> Plug.Conn.put_req_header("user-agent", "Mozilla/5.0")
+      |> Plug.Conn.put_req_header("referer", "http://www.example2.com/")
+      |> Plug.Conn.put_req_header("x-forwarded-for", "")
+      |> Plug.Conn.send_resp(200, "Hi!")
+
+    Logger.metadata(conn: conn)
+
+    log_entry =
+      capture_log(fn ->
+        Logger.debug("Hello")
+      end)
+      |> decode_or_print_error()
+
+    assert %{
+             "client.ip" => "",
+             "http.version" => "HTTP/1.1",
+             "http.request.method" => "GET",
+             "http.request.referrer" => "http://www.example2.com/",
+             "http.response.status_code" => 200,
+             "url.path" => "/",
+             "user_agent.original" => "Mozilla/5.0"
+           } = log_entry
+  end
+
+  test "logs exception http context" do
```

Comment:
Please check what additional tests were added to other loggers, I recently got "100% coverage" and would like to keep that :).

Reply:
Done, with 100% coverage!